spec/inputs/http_poller_spec.rb

```diff
@@ -23,7 +23,11 @@
   }
   let(:klass) { LogStash::Inputs::HTTP_Poller }
 
-  subject { klass.new(default_opts) }
+  subject {
+    s = klass.new(default_opts)
+    s.register # We can't use a before block here because of rspec scoping
```

Comment:
ahh, I think in that case it may make more sense to slightly refactor and move this subject up to a different level, like this pseudoexample:

```
# Currently:
describe foo do
  subject { first... }
  it "should stuff" do ... end
  describe bar do
    subject { second... }
    ...
  end
end

# Possibly better
describe foo do
  context "with some thing" do
    subject { first ... }
    it "should stuff" do ... end
  end

  describe bar do
    subject { second... }
  end
end
```

Basically, if you're going to shadow/override a let or subject deeper in your rspec examples, I think it's better to avoid that and create a new level to specifically avoid shadowing/overriding.
    
```